ironfish/src/utils/array.test.ts

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { ArrayUtils } from './array'
+
+describe('ArrayUtils', () => {
+  it('shuffles array', () => {
+    let items: number[] = []
+
+    for (let i = 0; i <= 10000; ++i) {
+      items.push(i)
+    }
+
+    items = ArrayUtils.shuffle(items)
+
+    const shuffled = items[0] !== 0 && items[500] !== 500 && items[1000] !== 1000
+    expect(shuffled).toBe(true)
+  })
+
+  it('sample a random item', () => {
+    let sample = ArrayUtils.sample([0, 1, 2])
+    expect(sample).toBeGreaterThanOrEqual(0)
+    expect(sample).toBeLessThanOrEqual(2)
```

Comment:
Similar problem to above: this test can pass if `ArrayUtils.sample()` returns always the first value. A better test would be to sample many values from the same array, and check that after ~100 or ~1000 samples, all possible samples have been returned.

Reply:
Improved this with the suggestion